tests/unit/test_base_service.py

```diff
@@ -254,3 +265,215 @@ def test_increment_interaction_count_repeated_on_multiple_scenarios(service: Bas
+    expected = [
+        Interaction(
+            name='interaction_0',
+            request=MITMRequest(
+                url="https://host_a.local/api",
+                method="GET",
+                headers=MITMHeaders(
+                    headers={
+                        'Accept': ['application/json'],
+                        'Accept-Encoding': ['gzip, deflate'],
+                        'Connection': ['keep-alive'],
+                        'User-Agent': ['python-requests/2.18.4']
+                    }
+                ),
+            ),
+            response=MITMResponse(
+                status_code=200,
+                headers=MITMHeaders(
+                    headers={
+                        'Content-Type': ['application/json; charset=UTF-8']
+                    }
+                ),
+                body=b'{"value": "response1"}'
+            ),
+        ),
+        Interaction(
+            name='interaction_1',
+            request=MITMRequest(
+                url="https://host_a.local/api",
+                method="POST",
+                headers=MITMHeaders(
+                    headers={
+                        'Accept': ['application/json'],
+                        'Accept-Encoding': ['gzip, deflate'],
+                        'Connection': ['keep-alive'],
+                        'User-Agent': ['python-requests/2.18.4']
+                    }
+                ),
+                body=b'{"value": "response2"}'
+            ),
+            response=MITMResponse(
+                status_code=201,
+                headers=MITMHeaders(
+                    headers={
+                        'Content-Type': ['application/json; charset=UTF-8']
+                    }
+                ),
+            ),
+        ),
+    ]
+
+    result = service.get_interactions_in_scenario(
+        Path(request.fspath.join('../fixtures/scenarios/test_scenario/TestService/'))
+    )
+
+    assert expected == result
+
+
+def test_get_interactions_for_active_scenario(service: BaseService, scenarios_dir):
+    expected = [
+        Interaction(
+            name='interaction_0',
+            request=MITMRequest(
+                url="https://host_a.local/api",
+                method="GET",
+                headers=MITMHeaders(
+                    headers={
+                        'Accept': ['application/json'],
+                        'Accept-Encoding': ['gzip, deflate'],
+                        'Connection': ['keep-alive'],
+                        'User-Agent': ['python-requests/2.18.4']
+                    }
+                ),
+            ),
+            response=MITMResponse(
+                status_code=200,
+                headers=MITMHeaders(
+                    headers={
+                        'Content-Type': ['application/json; charset=UTF-8']
+                    }
+                ),
+                body=b'{"value": "response1"}'
+            ),
+        ),
```

Comment:
Maybe use mocks or something nicer instead of full interaction objects (that way we decouple these tests from the Interaction objects themselves)